vm_args.c

```diff
@@ -170,6 +170,22 @@ args_rest_array(struct args_info *args)
     return ary;
 }
 
+static inline VALUE
+args_rest_array2(struct args_info *args)
+{
+    VALUE ary;
+
+    if (args->rest) {
+	ary = args->rest;
+	args->rest = 0;
+    }
+    else {
+	ary = GET_VM()->empty_args;
+	RB_GC_GUARD(ary);
```

Comment:
@nobu I thought it was unnecessary too, but the tests would segv without it.  I'll try removing it again.